src/scores/continuous/interval_impl.py

```diff
@@ -0,0 +1,226 @@
+            f"lower_qtile_level = {lower_qtile_level} and upper_qtile_level = {upper_qtile_level}"
+        )
+    # Check fcst_lower_qtile and fcst_upper_qtile have the same dimension
+    check_dims(fcst_upper_qtile, fcst_lower_qtile.dims, mode="equal")
+    specified_dims = reduce_dims or preserve_dims
+    # check requested dims are a subset of fcst_lower_qtile (or fcst_upper_qtile) dimensions
+    if specified_dims is not None:
+        check_dims(xr_data=fcst_lower_qtile, expected_dims=specified_dims, mode="superset")
+    # check obs dimensions are a subset of fcst_lower_qtile (or fcst_upper_qtile) dimensions
+    check_dims(xr_data=obs, expected_dims=fcst_lower_qtile.dims, mode="subset")  # type: ignore
+    if (fcst_lower_qtile > fcst_upper_qtile).any():
+        raise ValueError("Input does not satisfy fcst_lower_qtile < fcst_upper_qtile condition.")
+    interval_width = fcst_upper_qtile - fcst_lower_qtile
+    lower_diff = (1 / lower_qtile_level) * (fcst_lower_qtile - obs)
+    lower_diff = lower_diff.clip(min=0.0)
+    upper_diff = (1 / (1 - upper_qtile_level)) * (obs - fcst_upper_qtile)
+    upper_diff = upper_diff.clip(min=0.0)
+    total_score = interval_width + lower_diff + upper_diff
+    components = {
+        "interval_width_penalty": interval_width,
+        "overprediction_penalty": lower_diff,
+        "underprediction_penalty": upper_diff,
+        "total": total_score,
+    }
+    result = xr.Dataset(components)
+    reduce_dims = gather_dimensions(fcst_lower_qtile.dims, obs.dims, reduce_dims=reduce_dims, preserve_dims=preserve_dims)  # type: ignore[assignment]
+    results = apply_weights(result, weights=weights)
+    score = results.mean(dim=reduce_dims)
+    return score
+
+
+def interval_score(
+    fcst_lower_qtile: xr.DataArray,
+    fcst_upper_qtile: xr.DataArray,
+    obs: xr.DataArray,
+    interval_range: float,
+    *,  # Force keywords arguments to be keyword-only
+    reduce_dims: Optional[FlexibleDimensionTypes] = None,
+    preserve_dims: Optional[FlexibleDimensionTypes] = None,
+    weights: Optional[xr.DataArray] = None,
+) -> xr.Dataset:
+    """
+    Calculates the interval score for interval forecasts.
+    This function calls the `quantile_interval_score` function to calculate the interval score
+    for cases where quantile level range is symmetric.
+
+    .. math::
+    \\text{interval score} = (q_{u} - q_{l}) +
+    \\frac{2}{\\alpha} \\cdot (q_{l} - y) \\cdot \\mathbb{1}(y < q_{l}) + \\frac{2}{\\alpha} \\cdot (q_{u} - y) \\cdot \\mathbb{1}(y < q_{u})
+
+    where
+        - :math: `q_{u}` is the forecast at upper quantile
+        - :math: `q_{l}` is the forecast at lower quantile
+        - :math: `\\alpha` is confidence level that is equal to :math:`1 - interval_range`
+        - :math: `y` is the observation
+        - :math: `\\mathbb{1}(condition)` is an indicator function that is 1 when the condition is true, and 0 otherwise.
```

Comment:
```suggestion
    .. math::
        \\text{interval score} = (q_{u} - q_{l}) +
        \\frac{2}{\\alpha} \\cdot (q_{l} - y) \\cdot \\mathbb{1}(y < q_{l}) + \\frac{2}{\\alpha} \\cdot (q_{u} - y) \\cdot \\mathbb{1}(y < q_{u})

    where
        - :math:`q_{u}` is the forecast at upper quantile
        - :math:`q_{l}` is the forecast at lower quantile
        - :math:`\\alpha` is confidence level that is equal to :math:`1 - \\text{interval_range}`
        - :math:`y` is the observation
        - :math:`\\mathbb{1}(condition)` is an indicator function that is 1 when the condition is true, and 0 otherwise.
```